src/graphql/resolvers/__tests__/user.ts

```diff
@@ -0,0 +1,254 @@
+
+describe('Unit', () => {
+
+  describe('[Resolvers - Mutations]', () => {
+    
+
+    describe('user', () => {
+      
+
+      it('read one by Id - success', async () => {
+        mock.db.scan.mockReturnValueOnce({
+          promise: () => Promise.resolve(mockedReadByIdDbResponse)
+        });
+        const getNewId = jest.fn().mockImplementation(() => mockedDocumentId);
+        dataStorage.getNewId = getNewId.bind(dataStorage); 
+
+        await expect(userQueries.users(null, mockedReadByIdInputArgs, { dataSources: { storage: dataStorage}}, null)).resolves.toEqual(mockedReadByResponse);
+      });
+
+      it('read one by Id - failure', async () => {
+        mock.db.scan.mockReturnValueOnce({
+          promise: () => Promise.reject(new Error("read failed"))
+        });
+
+        await expect(userQueries.users(null, mockedReadByIdInputArgs, { dataSources: { storage: dataStorage}}, null)).rejects.toEqual(new ApolloError("read failed"));
+      });
+
+      it('read limited list - success', async () => {
+        mock.db.scan.mockReturnValueOnce({
+          promise: () => Promise.resolve(mockedReadLimitedListDbResponse)
+        });
+
+        await expect(userQueries.users(null, mockedReadLimitedListInputArgs, { dataSources: { storage: dataStorage}}, null)).resolves.toEqual(mockedReadLimitedListResponse);
+      });
+
+      it('read limited list - failure', async () => {
+        mock.db.scan.mockReturnValueOnce({
+          promise: () => Promise.reject(new Error("read failed"))
+        });
+
+        await expect(userQueries.users(null, mockedReadLimitedListInputArgs, { dataSources: { storage: dataStorage}}, null)).rejects.toEqual(new ApolloError("read failed"));
+      });
+    });
+    
+    describe('create service', () => {
+      it('success', async () => {
+        mock.db.put.mockReturnValueOnce({
+          promise: () => Promise.resolve()
+        });
+
+        await expect(userMutations.createUser(null, mockedCreateInputArgs, { dataSources: { storage: dataStorage}}, null)).resolves.toEqual(mockedCreateResponse);
+      });
+
+      it('failure', async () => {
+        mock.db.put.mockReturnValueOnce({
+          promise: () => Promise.reject(new Error("creation failed"))
+        });
+
+        await expect(userMutations.createUser(null, mockedCreateInputArgs, { dataSources: { storage: dataStorage}}, null)).rejects.toEqual(new ApolloError("creation failed"));
+      });
+    });
+
+    describe('update service', () => {
+      it('success', async () => {
+        mock.db.update.mockReturnValueOnce({
+          promise: () => Promise.resolve(mockedUpdateResponse)
+        });
+
+        await expect(userMutations.updateUser(null, mockedUpdateInputArgs, { dataSources: { storage: dataStorage}}, null)).resolves.toEqual(mockedUpdateResponse.Attributes);
+      });
+
+      it('failure', async () => {
+        mock.db.update.mockReturnValueOnce({
+          promise: () => Promise.reject(new Error("Document not found"))
```

Comment:
Does this error (`new Error("Document not found")`) need to be specified in order for the following line to pass? Or can we pass any error and expect the same behavior?

`await expect(userMutations.updateUser(null, mockedUpdateInputArgs, { dataSources: { storage: dataStorage}}, null)).rejects.toEqual(new ApolloError("Document not found"));`

Reply:
Yes, it should specified in this way, because I decide to catch errors comming from the db if the id doesn't exist and reject the promise with this particular error message.

This is not reusable for other exceptions.

going further, the catch could have more inteligence, and based on the evaluation of the exception, to throw different masked messages. 

Or maybe just throw the oroginal exception. Anyway, it will depend the design that we do.